sys/fatfs_diskio/sdcard_spi/sdcard_spi_diskio.c

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (C) 2016 Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup  tests
+ * @{
+ *
+ * @file
+ * @brief    Implementation of fatfs interface that supports sdcard_spi driver
+ *           based on low level disk I/O module example for FatFs by ChaN, 2016
+ *
+ * @author   Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ *
+ * @}
+ */
+#include "fatfs/diskio.h"       /* FatFs lower layer API */
+#include "fatfs/integer.h"
+#include "sdcard_spi.h"
+
+#include <stdio.h>
+#include <time.h>
+#include "periph_conf.h"
+#include "periph/rtc.h"
+#include "xtimer.h"
+#include "debug.h"
+
+/* Complete pending write process (needed at _FS_READONLY == 0) */
+#define CTRL_SYNC           0   
+
+/* Get media size (needed at _USE_MKFS == 1) */
+#define GET_SECTOR_COUNT    1   
+
+/* Get sector size (needed at ) */
+#define GET_SECTOR_SIZE     2   
+
+/* Get erase block size (needed at _USE_MKFS == 1) */
+#define GET_BLOCK_SIZE      3   
+
+/* Inform device that the data on the block of sectors is 
+   no longer used (needed at _USE_TRIM == 1) */
+#define CTRL_TRIM           4   
+
+#define RTC_YEAR_OFFSET   1900
+#define FATFS_YEAR_OFFSET 1980
+
+bool rtc_init_done = false;
+
+extern sd_card_t cards[NUM_OF_SD_CARDS];
+
+static inline sd_card_t *get_sd_card(int idx){
+    if(idx < NUM_OF_SD_CARDS){
+        return &(cards[idx]);
+    }else{
+        return NULL;
+    }
+}
+
+#ifdef FATFS_RTC_AVAILABLE
+DWORD get_fattime (void){

```

Comment:
seems to be the same function as above. would make sense to put all common functions / macros into a common file